portality/api/v1/crud/articles.py

```diff
@@ -95,20 +95,26 @@ def retrieve_swag(cls):
         template['responses']['200']['schema']['properties'] = OutgoingArticleDO().struct_to_swag()
         template['responses']['401'] = cls.R401
         template['responses']['404'] = cls.R404
-        return cls._build_swag_response(template)
+        return cls._build_swag_response(template, api_key_optional_override=True)
 
     @classmethod
     def retrieve(cls, id, account):
-        # as long as authentication (in the layer above) has been successful, and the account exists, then
-        # we are good to proceed
-        if account is None:
-            raise Api401Error()
 
-        # is the article id valid
+        # is the article id valid?
         ar = models.Article.pull(id)
         if ar is None:
             raise Api404Error()
 
+        # at this point we're happy to return the article if it's
+        # meant to be seen by the public
+        if ar.get('admin').get('in_doaj', False):
```

Comment:
```ar.is_in_doaj()``` probably better idea. If you read it you'll see that it also assumes ```{}``` if the first get returns nothing, otherwise the second one will cause a 500 error. But seeing as a method already does it .. :)